HopsanGenerator/src/HopsanGeneratorLib.cpp

```diff
@@ -456,7 +483,7 @@ bool callAddComponentToLibrary(const char* libraryXmlPath, const char* targetPat
 
     //Add new component to library
     lib.mComponentXMLFiles.append(QDir(xmlPath.absolutePath()).relativeFilePath(QFileInfo(cafPath).absoluteFilePath()));
-    lib.mComponentCodeFiles.append(QFileInfo(hppPath).fileName());
+    lib.mComponentCodeFiles.append(QFileInfo(sourcePath).fileName().replace(".mo", ".hpp"));
```

Comment:
What if a c++ source file has  .mo in its name,   like:   my-super.mo-sourcefile.hpp
Replace if ends with .mo ?